kolena/_experimental/dataset/_dataset.py

```diff
@@ -217,23 +232,17 @@ def register_dataset(
     """
     load_uuid = init_upload().uuid
     existing_dataset = load_dataset(name)
-    existing_id_fields = []
-    if existing_dataset:
-        existing_id_fields = existing_dataset.id_fields
-    if not id_fields:
-        if existing_id_fields:
-            raise InputValidationError("id_fields is required for updating an existing dataset")
-        else:
-            id_fields = _infer_id_fields(df)
     if isinstance(df, pd.DataFrame):
-        validate_id_fields(df, id_fields, existing_id_fields)
+        id_fields = resolve_id_fields(df, id_fields, existing_dataset)
+        validate_dataframe_ids(df, id_fields)
         _upload_dataset_chunk(df, load_uuid, id_fields)
     else:
         validated = False
         for chunk in df:
             if not validated:
+                id_fields = resolve_id_fields(chunk, id_fields, existing_dataset)
```

Comment:
cc @grant-Kolena for reviewing this change. I noticed original logic could break when iterator is passed